File: solutions/csharp/resistor-color-trio/1/ResistorColorTrio.cs
using System;
using System.Collections.Generic;

public static class ResistorColorTrio
{
    private static Dictionary<string, int> _colors = new Dictionary<string, int> {
        { "black", 0 },
        { "brown", 1 },
        { "red", 2 },
        { "orange", 3 },
        { "yellow", 4 },
        { "green", 5 },
        { "blue", 6 },
        { "violet", 7 },
        { "grey", 8 },
        { "white", 9 }
    };
    
    public static string Label(string[] colors)
    {
        int number = 0;
        foreach (string color in colors[..2])
        {
            number = 10 * number + _colors[color];
        }
        number *= (int)Math.Pow((double)10, (double)_colors[colors[2]]);
        
        if (number > 1000)
        {
            return $"{number / 1000} kiloohms";
        }
        else
        {
            return $"{number} ohms";
        }
    }
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
"""A simple function for decoding resistor color bands"""

def label(colors):
    color_list = ["black", 
                 "brown", 
                 "red", 
                 "orange", 
                 "yellow", 
                 "green", 
                 "blue", 
                 "violet", 
                 "grey", 
                 "white"]
    ohms = (10 * color_list.index(colors[0]) \
        + color_list.index(colors[1])) \
        * 10 ** color_list.index(colors[2])
    if ohms >= 1000000000:
        return str(int(ohms / 1000000000)) + " gigaohms"
    if ohms >= 1000000:
        return str(int(ohms / 1000000)) + " megaohms"
    if ohms >= 1000:
        return str(int(ohms / 1000)) + " kiloohms"
    return str(ohms) + " ohms"
